### PR TITLE
IsolateEmulatorThread: Complete to even parity

### PR DESCRIPTION
**Pull request by orelmisan**
PR kubevirt/kubevirt#10872 added a cluster-wide setting that makes it possible to run
VMs with dedicated CPU placement and emulator thread isolation on compute nodes with SMT [1] enabled and `full-pcpus-only` CPUManager policy [2].

This configuration is used for high-performance VMs.

Add an explanation on how to use it.

[1] https://en.wikipedia.org/wiki/Simultaneous_multithreading
[2] https://kubernetes.io/docs/tasks/administer-cluster/cpu-management-policies/#static-policy-options

### CODE REVIEW

**Review by RamLavi — APPROVED** [review]
/lgtm

**Comment by orelmisan** [comment]
kubevirt/kubevirt#10872 was merged.
/hold cancel

**Review by xpivarc — APPROVED** [review]

File: docs/virtual_machines/dedicated_cpu_resources.md [hunk 1]
```diff
@@ -114,6 +114,26 @@ Example:
...
+```
+
+In order to address this issue:
+1. Enable the `AlignCPUs` feature gate in the KubeVirt CR.
+2. Add the following annotation to the Kubevirt CR:
+```yaml
+alpha.kubevirt.io/EmulatorThreadCompleteToEvenParity:
+```
+
+It will cause KubeVirt to add one or two dedicated CPUs for the emulator threads, in a way that completes the total CPU count to be even.
```
[the remark this answers is not in the document]
Comment (orelmisan) [reply docs/virtual_machines/dedicated_cpu_resources.md]: Done.

File: docs/virtual_machines/dedicated_cpu_resources.md [hunk 2]
```diff
@@ -114,6 +114,26 @@ Example:
             cpu: 2
             memory: 2Gi
 
+When the following conditions are met:
```
[the remark this answers is not in the document]
Comment (orelmisan) [reply docs/virtual_machines/dedicated_cpu_resources.md:119]: Added a subheading `Compute Nodes with SMT Enabled`.

**Comment by orelmisan** [comment]
[Change](https://github.com/kubevirt/user-guide/compare/17f86a85c9dccbab4106daf57d77ecbf62c95824..ab424efd2a5da30130a47167c42c36f9a770a6b8): Answered review [comments](https://github.com/kubevirt/user-guide/pull/745#pullrequestreview-1809296540).

**Pull request merged**